src/main/java/com/zufar/onlinestore/payment/processor/PaymentEventProcessor.java

```diff
@@ -0,0 +1,44 @@
+package com.zufar.onlinestore.payment.processor;
+
+import com.stripe.exception.SignatureVerificationException;
+import com.stripe.model.Event;
+import com.stripe.net.Webhook;
+import com.zufar.onlinestore.payment.config.StripeConfiguration;
+import com.zufar.onlinestore.payment.enums.PaymentStatus;
+import com.zufar.onlinestore.payment.model.Payment;
+import com.zufar.onlinestore.payment.repository.PaymentRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class PaymentEventProcessor {
+
+    private final StripeConfiguration stripeConfig;
+    private final PaymentRepository paymentRepository;
+
+    public Event processEvent(String payload, String header) throws SignatureVerificationException {
+        return Webhook.constructEvent(payload, header, stripeConfig.webHookSecretKey());
+    }
+
+    public void processSucceededPaymentIntent(String paymentIntentId) {
+        updateDataTemplateMethod(paymentIntentId, PaymentStatus.SUCCEEDED);
+    }
+
+    public void processHandleFailedPaymentIntent(String paymentIntentId) {
+        updateDataTemplateMethod(paymentIntentId, PaymentStatus.PAYMENT_FAILED);
+    }
+
+    private void updateDataTemplateMethod(String paymentIntentId, PaymentStatus status) {
```

Comment:
`updateDataTemplateMethod` - method is not typo?

Reply:
> `updateDataTemplateMethod` - method is not typo?

Done